tools/downloader/topologies.py

```diff
@@ -0,0 +1,196 @@
+import sys
+import os
+import re
+import yaml
+import types
+import argparse
+
+from pathlib import Path
+
+import common
+
+_precision_marker = '$precision'
+_cache = Path(os.getenv('OPENCV_OPEN_MODEL_ZOO_CACHE_DIR', Path('.').resolve() / 'models'))
```

Comment:
User-defined directory can be used (`OPENCV_OPEN_MODEL_ZOO_CACHE_DIR` environment variable).

Reply:
There should be a reasonable fallback folder in case `OPENCV_OPEN_MODEL_ZOO_CACHE_DIR` is not defined. Could be created by `tempfile.mkdtemp()`. 

Why is the `OPENCV_OPEN_MODEL_ZOO_CACHE_DIR` prefixed with `OPENCV`? This pull request is for an OpenVINO module. I would just drop the prefix, or prefix with `OPENVINO`.

Reply:
> There should be a reasonable fallback folder in case OPENCV_OPEN_MODEL_ZOO_CACHE_DIR is not defined. Could be created by tempfile.mkdtemp().

There is a fallback already - folder named `models` in the current working directory.

> Why is the OPENCV_OPEN_MODEL_ZOO_CACHE_DIR prefixed with OPENCV? This pull request is for an OpenVINO module. I would just drop the prefix, or prefix with OPENVINO.

Renamed it to just `OPEN_MODEL_ZOO_CACHE_DIR`. `OPENVINO_OPEN_MODEL_ZOO_CACHE_DIR` is to long :)